CMakeLists.txt

```diff
@@ -791,6 +791,8 @@ if(IREE_BUILD_BENCHMARKS OR IREE_BUILD_EXPERIMENTAL_E2E_TEST_ARTIFACTS)
   endif()
 endif()
 
+add_subdirectory(testing)
```

Comment:
We don't want a new top-level directory here. We already have tests/

Reply:
Isn't `tests/` for the tests themselves? This is a Python module that contains functionality used by tests. Like `numpy.testing`, but `iree.testing` instead. Where is it appropriate to put it?

Reply:
Ah, fair. I'm still a bit skeptical of adding a new top-level directory basically just for this one test that is pretty unusual in the context of the repo.

@ScottTodd thoughts?

Reply:
We definitely don't want a new top level directory for a single test (or helper functions for a single test, in this case). The simplest solution is to move the `assert_array_almost_equal` and other helper functions into the new test file itself. Once the functions are used by more than one test we can find a shared place for them.

That does bring up another point though - I'm not sure if we want this test to be written in Python at all, since then it can only run on platforms with a Python interpreter. We could use either `iree_check_test` or `iree_run_module_test` instead.
* We have Python tests for the Python bindings here: https://github.com/iree-org/iree/tree/main/runtime/bindings/python/tests
* We have Python tests for (TF-based) Python frameworks here: https://github.com/iree-org/iree/tree/main/integrations/tensorflow/test/python 

To the other comment thread about GCS files, doesn't `iree_run_module_test` already handle those a bit?

Reply:
I moved the stuff from `testing` into the test itself.